xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -0,0 +1,164 @@
+from collections.abc import Sequence
+from typing import Any, cast
+
+from xdsl.backend.riscv.lowering.utils import (
+    cast_operands_to_regs,
+    register_type_for_type,
+)
+from xdsl.dialects import memref, riscv
+from xdsl.dialects.builtin import (
+    Float32Type,
+    IntegerType,
+    ModuleOp,
+    UnrealizedConversionCastOp,
+)
+from xdsl.ir import MLContext, Operation, SSAValue
+from xdsl.ir.core import Attribute
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.exceptions import DiagnosticException
+
+
+class ConvertMemrefAllocOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref.Alloc, rewriter: PatternRewriter) -> None:
+        raise DiagnosticException("Lowering memref.alloc not implemented yet")
+
+
+class ConvertMemrefDeallocOp(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: memref.Dealloc, rewriter: PatternRewriter) -> None:
+        raise DiagnosticException("Lowering memref.dealloc not implemented yet")
+
+
+def memref_shape_ops(
+    mem: SSAValue,
+    indices: Sequence[SSAValue],
+    shape: Sequence[int],
+    element_type: Attribute,
+) -> tuple[list[Operation], SSAValue]:
+    """
+    Returns ssa value representing pointer into the memref at given indices.
+    The pointer is byte-indexed, and the indices are strided by element size, so the index
+    into the flat memory buffer needs to be multiplied by the size of the element.
+    """
+
+    assert len(shape) == len(indices)
+
+    # Only handle a small subset of elements
+    # Might be useful as a helper for other passes in the future
+    match element_type:
+        case IntegerType():
```

Comment:
I think it's better to constrain this to int32, since using int8 with this method might fail, I think.
If you want to access a int8 integer, you need to access the entire 4-bytes and then shift until you get the right number, I believe, because riscv accesses are four bytes aligned, I think?
Or you should start using other instructions, like SB and LB, I think?
What do you think @superlopuh ?

Reply:
Sure sounds good I'll add a special case for i32